docs/topics/generics.md

```diff
@@ -322,6 +322,73 @@ fun <T> copyWhenGreater(list: List<T>, threshold: T): List<String>
 The passed type must satisfy all conditions of the `where` clause simultaneously. In the above example, the `T` type
 must implement _both_ `CharSequence` and `Comparable`.
 
+## Definitely non-nullable types
+
+To make interoperability with generic Java classes and interfaces easier, Kotlin supports declaring a generic type parameter
+as **definitely non-nullable**. 
+
+To declare a generic type `T` as definitely non-nullable, declare the type with `& Any`. For example: `T & Any`. 
+
+> The syntax represents an intersection between type `T` and the non-nullable type `Any`.
+> 
+{type="note"}
+
+A definitely non-nullable type must have a nullable [upper bound](#upper-bounds). In addition, `Any` must resolve to `kotlin.Any`.
```

Comment:
> In addition, `Any` must resolve to `kotlin.Any`

IMO, this sentence may create unnecessary confusion. In the rest of the page, it's just `Any` without `kotlin.Any`. It may leave questions like 'what else Any can resolve to?' without providing a solid answer or a link to that answer.